### PR TITLE
Start a Compatibility51 library for backporting fixes to Swift 5.1 runtimes

### PR DESCRIPTION
**Pull request by jckarter**

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
cc @Azoy, hopefully this won't interfere with your work on tuple conformances too much!

File: lib/Driver/DarwinToolChains.cpp [hunk 1]
```diff
@@ -367,6 +367,18 @@ toolchains::Darwin::addArgsToLinkStdlib(ArgStringList &Arguments,
         Arguments.push_back(context.Args.MakeArgString(BackDeployLib));
       }
     }
+
+    if (*runtimeCompatibilityVersion <= llvm::VersionTuple(5, 1)) {
```
Comment (Azoy) [inline lib/Driver/DarwinToolChains.cpp:371]: There's another version check slightly above this for the `runtime-compatibility-version` flag. Should that be updated as well?

File: stdlib/toolchain/Compatibility51/Overrides.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//===--- Overrides.cpp - Compat override table for Swift 5.1 runtime ------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
```
Comment (Azoy) [inline stdlib/toolchain/Compatibility51/Overrides.cpp:5]: 2020 here?

**Review by mikeash — APPROVED** [review]

File: lib/Driver/DarwinToolChains.cpp (hunk 1 above)
Comment (jckarter) [reply lib/Driver/DarwinToolChains.cpp:371]: Indeed, thanks!

**Comment by jckarter** [comment]
@swift-ci Please test

**Review by aschwaighofer — APPROVED** [review]

**Comment by jckarter** [comment]
@swift-ci Please test Linux

**Pull request merged**